### PR TITLE
Optionally run steps as subprocesses

### PR DESCRIPTION
**Pull request by xylar**
This will enable calls to external packages that may, in turn, call subprocess.  If this functionality is called directly, there is no obvious way to capture stdout/stderr in a log file.  One example of this is jigsawpy's functionality for creating icosahedral meshes.

### CODE REVIEW

**Comment by xylar** [comment]
@matthewhoffman, all I'm looking for in a review here is:
* making sure you're okay with this change
* a quick MALI test to make sure this does no harm.

It would be great to get this in soonish because I want to build on it (draft PR coming later this weekend).

**Comment by xylar** [comment]
## Testing

I have tested this using my https://github.com/xylar/compass/tree/separate_mesh_steps branch.  Previously (without using `subprocess`) when I created icosahedral meshes, JIGSAW output was going to the terminal.  With this fix, it goes to the log file as desired.

File: docs/developers_guide/organization.rst [hunk 1]
```diff
@@ -854,6 +854,13 @@ Some attributes are available after calling the base class' constructor
     Whether to get all of the outputs for the step from the database of
     cached outputs for the MPAS core that this step belongs to
 
+``self.run_as_subprocess``
```
Comment (matthewhoffman) [inline docs/developers_guide/organization.rst:857]: @xylar , can you elaborate on how running as a subprocess or not changes the logging behavior?  Is subprocess stdout lost, sent to the terminal/job log, or something else?

Comment (xylar) [reply docs/developers_guide/organization.rst:857]: @matthewhoffman Do you want me to elaborate on that here or in the documentation?

**Review by matthewhoffman — APPROVED** [review]
I ran our integration suite with this branch and observed no ill effects.  I'm approving and leaving to your judgment whether and how to adjust the docs per the thread on that.

**Comment by xylar** [comment]
Thanks, @matthewhoffman!  I did indeed add quite a bit more detail to the docs.

**Pull request merged**